[github-actions] build checks fail intermittently

Log snippet:
```
++ repo_dir=/home/runner/work/ot-efr32/ot-efr32
+ OT_OPTIONS=("-DCMAKE_BUILD_TYPE=MinSizeRel" "-DOT_DIAGNOSTIC=ON" "-DOT_EXTERNAL_HEAP=ON" "-DOT_SLAAC=ON")
+ readonly OT_OPTIONS
+ main efr32mg13 -DOT_COMPILE_WARNING_AS_ERROR=ON -DOT_COMMISSIONER=ON -DOT_DHCP6_CLIENT=ON -DOT_DHCP6_SERVER=ON -DOT_DNS_CLIENT=ON -DOT_JOINER=ON -DOT_BORDER_ROUTER=ON -DOT_REFERENCE_DEVICE=ON -DOT_COAP=ON -DOT_COAPS=ON -DOT_LOG_LEVEL_DYNAMIC=ON -DOT_DIAGNOSTIC=ON -DOT_FULL_LOGS=ON -DOT_MAC_FILTER=ON -DBOARD=brd4168a
+ local 'usage=usage: /home/runner/work/ot-efr32/ot-efr32/script/build [-h] [--skip-silabs-apps] <efr32 platform> -DBOARD=<brdXXXXy> [-D<OT_XXXX=ON> -D<OT_YYYY=OFF>]'
+ local skip_silabs_apps=false
+ optspec=:h-:
+ getopts :h-: optchar
+ local efr32_platforms
++ efr32_get_platforms
++ '[' 0 -ne 0 ']'
++ local platform_dir=/home/runner/work/ot-efr32/ot-efr32/src
++ find /home/runner/work/ot-efr32/ot-efr32/src -type d -name '*efr32*' -exec basename '{}' ';'
++ sort
+ efr32_platforms='efr32mg1
efr32mg12
efr32mg13
efr32mg21'
+ [[ 16 == 0 ]]
+ local platform=efr32mg13
+ grep -wq efr32mg13
+ echo 'efr32mg1
efr32mg12
efr32mg13
efr32mg21'
/home/runner/work/ot-efr32/ot-efr32/script/build: line 128: echo: write error: Broken pipe
+ die 'Unsupported platform efr32mg13'
 ** ERROR: Unsupported platform efr32mg13
+ echo ' ** ERROR: Unsupported platform efr32mg13'
+ exit 1
Error: Process completed with exit code 1.
```

[logs_566.zip](https://github.com/openthread/ot-efr32/files/8072901/logs_566.zip)
https://github.com/openthread/ot-efr32/runs/5195633403?check_suite_focus=true

--- a/script/build
+++ b/script/build
@@ -125,7 +125,7 @@ main()
     fi
 
     local platform="$1"
-    echo "$efr32_platforms" | grep -wq "${platform}" || die "Unsupported platform ${platform}"
+    efr32_check_platform "${platform}" || die "Unsupported platform ${platform}"
     shift
 
     local options=("${OT_OPTIONS[@]}")

